tfmigrate/state_replace_provider_action_test.go

```diff
@@ -0,0 +1,38 @@
+package tfmigrate
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/minamijoyo/tfmigrate/tfexec"
+)
+
+func TestAccStateReplaceProviderAction(t *testing.T) {
+	tfexec.SkipUnlessAcceptanceTestEnabled(t)
+
+	backend := tfexec.GetTestAccBackendS3Config(t.Name())
+
+	source := `
+resource "null_resource" "foo" {}
+`
+
+	workspace := "default"
+	tf := tfexec.SetupTestAccWithApply(t, workspace, backend+source)
+	ctx := context.Background()
+
+	actions := []StateAction{
+		NewStateReplaceProviderAction("registry.terraform.io/hashicorp/null", "registry.tfmigrate.io/hashicorp/null"),
+	}
+
+	m := NewStateMigrator(tf.Dir(), workspace, actions, &MigratorOption{}, false)
+	err := m.Plan(ctx)
+	if err == nil {
+		t.Fatalf("expected migrator plan error but received: %s", err)
+	}
+
+	expected := "Could not load the schema for provider\nregistry.tfmigrate.io/hashicorp/null"
+	if !strings.Contains(err.Error(), expected) {
+		t.Fatalf("expected migrator plan error to include: %s", expected)
```

Comment:
☝️ There may be a better way to author this test. In this case, because we've replaced the legitimate `registry.terraform.io/hashicorp/local` provider with an illegitimate `registry.tfmigrate.io/hashicorp/local` provider `plan` is expected to error, but we can verify that it does so with the expected illegitimate provider replacement.